src/compiler/scala/tools/nsc/transform/LazyVals.scala

```diff
@@ -100,11 +100,26 @@ abstract class LazyVals extends Transform with TypingTransformers with ast.TreeD
             val (rhs1, sDef) = mkLazyDef(enclosingClassOrDummyOrMethod, transform(rhs), idx, sym)
             sym.resetFlag((if (lazyUnit(sym)) 0 else LAZY) | ACCESSOR)
             (rhs1, sDef)
-          } else
+          } else if (sym.hasAllFlags(MODULE | METHOD) && !sym.owner.isTrait) {
```

Comment:
Some comment would be helpful here. Maybe you could also extend a general doc comment somewhere (on `transform`?) to explain a bit more what's going on - eg what are "slowPath" and "fastPath"?